src/libos/crates/jindisk/src/index/mem_table.rs

```diff
@@ -0,0 +1,214 @@
+//! MemTable of lsm tree.
+use crate::prelude::*;
+use crate::util::RangeQueryCtx;
+use crate::Record;
+use async_rt::sync::RwLockWriteGuard;
+
+use std::collections::BTreeMap;
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+pub type BTreeMemTable = BTreeMap<Lba, MemRecord>;
+// TODO: Add `RBTreeMemTable`
+
+/// A thread-safe `MemTable`.
+pub struct LockedMemTable {
+    mem_table: AsyncRwLock<BTreeMemTable>,
+    num_records: AtomicUsize,
+    capacity: usize,
+    reclaim_fn: Option<ReclaimFn>,
+}
+
+pub type ReclaimFn =
+    Box<dyn Fn(Record, MemRecord) -> Pin<Box<dyn Future<Output = MemRecord> + Send>> + Send + Sync>;
```

Comment:
Is it possible not to copy records? In other words, can we change the signature to the following form.

```rust
pub type ReclaimFn =
    Box<dyn Fn(&Record, &MemRecord) -> Pin<Box<dyn Future<Output = MemRecord> + Send>> + Send + Sync>;
```